src/client/pfs.go

```diff
@@ -362,37 +366,51 @@ func (c APIClient) DeleteBranch(repoName string, branchName string, force bool)
 	return grpcutil.ScrubGRPC(err)
 }
 
-// SquashCommit deletes a commit.
-func (c APIClient) SquashCommit(repoName string, branchName string, commitID string) error {
-	_, err := c.PfsAPIClient.SquashCommit(
+// InspectJob returns info about a specific Job.
+func (c APIClient) InspectJob(jobID string) (*pfs.JobInfo, error) {
+	resp, err := c.PfsAPIClient.InspectJob(
 		c.Ctx(),
-		&pfs.SquashCommitRequest{
-			Commit: NewCommit(repoName, branchName, commitID),
+		&pfs.InspectJobRequest{
+			Job: NewJob(jobID),
+		},
+	)
+	if err != nil {
+		return nil, grpcutil.ScrubGRPC(err)
+	}
+	return resp, nil
+}
+
+// SquashJob squashes the commits of a job into their children.
+func (c APIClient) SquashJob(jobID string) error {
+	_, err := c.PfsAPIClient.SquashJob(
+		c.Ctx(),
+		&pfs.SquashJobRequest{
+			Job: NewJob(jobID),
 		},
 	)
 	return grpcutil.ScrubGRPC(err)
 }
 
-// FlushCommit calls cb with commits that have the specified `commits` as
-// provenance. Note that it can block if jobs have not successfully
-// completed. This in effect waits for all of the jobs that are triggered by a
-// set of commits to complete.
+// FlushJob calls cb with commits that are part of the specified Job.  Note that
+// it can block if jobs have not successfully completed. This in effect waits
+// for all of the jobs that are triggered by a set of commits to complete.
 //
-// If toRepos is not nil then only the commits up to and including those repos
-// will be considered, otherwise all repos are considered.
+// If toBranches is not nil then only the commits up to and including those
+// branches will be considered, otherwise all branches affected by the job are
+// considered.
 //
-// Note that it's never necessary to call FlushCommit to run jobs, they'll run
-// no matter what, FlushCommitF just allows you to wait for them to complete and
+// Note that it's never necessary to call FlushJob to run PipelineJobs, they'll
+// run no matter what, FlushJob just allows you to wait for them to complete and
 // see their output once they do.
-func (c APIClient) FlushCommit(commits []*pfs.Commit, toRepos []*pfs.Repo, cb func(*pfs.CommitInfo) error) (retErr error) {
+func (c APIClient) FlushJob(job *pfs.Job, toBranches []*pfs.Branch, cb func(*pfs.CommitInfo) error) (retErr error) {
```

Comment:
Maybe this is a good opportunity to rename Flush. I think it's one of the more confusing verbs in our API, people often don't find it when they need it and some people assume they need it to make jobs run. Strawman proposal: `WaitJob`?

Reply:
`WaitJob` sounds good to me :+1:

Reply:
Are we going with Block or Wait? It looks like both verbs are being using currently (BlockCommitset vs the field being Wait).

Reply:
Good point, renamed the field to `Block`.